docs/index.md

```diff
@@ -0,0 +1,437 @@
+```
+
+#### As Infrastructure
+
+You can also use Cloud Run Jobs as infrastructure to execute your deployed flows.
+
+##### Set variables
+
+To expedite copy/paste without the needing to update placeholders manually, update and execute the following.
+
+```bash
+export CREDENTIALS_BLOCK_NAME="BLOCK-NAME-PLACEHOLDER"
+export CLOUD_RUN_JOB_BLOCK_NAME="cloud-run-job-example"
+export CLOUD_RUN_JOB_REGION="us-central1"
+export GCS_BUCKET_BLOCK_NAME="cloud-run-job-bucket-example"
+export GCP_PROJECT_ID=$(gcloud config get-value project)
+```
+
+##### Build an image
+
+First, find an existing image within the Google Artifact Registry. Ensure it has Python and `prefect-gcp[cloud_storage]` installed, or follow the instructions below to set one up.
+
+Create a `Dockerfile`.
+
+```dockerfile
+FROM prefecthq/prefect:2-python3.11
+RUN pip install "prefect-gcp[cloud_storage]"
+```
+
+Then push to the Google Artifact Registry.
+
+```bash
+gcloud artifacts repositories create test-example-repository --repository-format=docker --location=us
+gcloud auth configure-docker us-docker.pkg.dev
+docker build -t us-docker.pkg.dev/${GCP_PROJECT_ID}/test-example-repository/prefect-gcp:2-python3.11 .
+docker push us-docker.pkg.dev/${GCP_PROJECT_ID}/test-example-repository/prefect-gcp:2-python3.11
+```
+
+##### Save an infrastructure and storage block
+
+Save a custom infrastructure and storage block by executing the following snippet.
+
+```python
+import os
+from prefect_gcp import GcpCredentials, CloudRunJob, GcsBucket
+
+gcp_credentials = GcpCredentials.load(os.environ["CREDENTIALS_BLOCK_NAME"])
+
+# must be from GCR and have Python + Prefect
+image = f"us-docker.pkg.dev/{os.environ['GCP_PROJECT_ID']}/test-example-repository/prefect-gcp:2-python3.11"  # noqa
+
+cloud_run_job = CloudRunJob(
+    image=image,
+    credentials=gcp_credentials,
+    region=os.environ["CLOUD_RUN_JOB_REGION"],
+)
+cloud_run_job.save(os.environ["CLOUD_RUN_JOB_BLOCK_NAME"], overwrite=True)
+
+bucket_name = "cloud-run-job-bucket"
+cloud_storage_client = gcp_credentials.get_cloud_storage_client()
+cloud_storage_client.create_bucket(bucket_name)
+gcs_bucket = GcsBucket(
+    bucket=bucket_name,
+    gcp_credentials=gcp_credentials,
+)
+gcs_bucket.save(os.environ["GCS_BUCKET_BLOCK_NAME"], overwrite=True)
+```
+
+##### Write a flow
+
+Then, find an existing flow or define an arbitrary one to test.
+
+```python
+from prefect import flow
+
+@flow(log_prints=True)
+def cloud_run_job_flow():
+    print("Hello, Prefect!")
+
+if __name__ == "__main__":
+    cloud_run_job_flow()
+```
+
+##### Create a deployment
+
+If the script was named "cloud_run_job_script.py", build the deployment with the following command.
+
+```bash
+prefect deployment build cloud_run_job_script.py:cloud_run_job_flow \
+    -n cloud-run-deployment \
+    -ib cloud-run-job/${CLOUD_RUN_JOB_BLOCK_NAME} \
+    -sb gcs-bucket/${GCS_BUCKET_BLOCK_NAME}
+```
+
+Make any necessary changes to the YAML, if needed, and apply the deployment.
```

Comment:
If we're going to suggest changes to deployment manifest, we should have specific examples.
```suggestion
Now apply the deployment!
```